context/secrets.js

```diff
@@ -4,22 +4,33 @@ import React, {
   useMemo,
   useContext,
   useCallback,
+  useEffect,
 } from 'react'
 
-import { upsert, find } from '../lib/secretsManager'
+import SecretsManager from '../lib/secretsManager'
+
+const secretsManager = new SecretsManager()
+
+;(async () => await secretsManager.init())()
 
 const SecretsContext = createContext({})
 
-export function useSecretsContext() {
+export function useSecrets() {
   return useContext(SecretsContext)
 }
 
 export function SecretsProvider({ children }) {
   const [secrets, setSecrets] = useState([])
 
+  useEffect(() => {
+    ;(async () => {
+      setSecrets(await secretsManager.find())
+    })()
```

Comment:
avoid this pattern. if you need to run something async wrap it in a function and invoke it from within the useeffect hook

Reply:
The useEffect callback can't be an async function so the alternative I guess would be to use the promise syntax. Is that what you mean?

Reply:
Ok I guess it's my personal preference but I really do not like this iife pattern in useEffect. I would prefer something less exotic, which works the same:

```suggestion
    async function storeSecrets() {
      setSecrets(await secretsManager.find())
    }
    storeSecrets()
```

Reply:
That's fine too! I'll change it!